lib/src/signer.rs

```diff
@@ -0,0 +1,51 @@
+#[cfg(feature = "signing-gpgme")]
+use gpgme::{Context, Key, Protocol};
+
+use crate::backend::BackendResult;
+
+pub trait Signer {
+    fn sign(&self, input: &[u8]) -> BackendResult<String>;
+}
+
+#[cfg(feature = "signing-gpgme")]
+pub struct GpgMeSigner {
+    key: Option<Key>,
+}
+
+#[cfg(feature = "signing-gpgme")]
+impl GpgMeSigner {
+    pub fn new(key_id: Option<String>) -> Self {
+        let key = match key_id {
+            Some(key_id) => {
+                let mut ctx =
+                    Context::from_protocol(Protocol::OpenPgp).expect("from_protocol failed");
+                let mut keys = ctx.find_keys([key_id]).expect("failed to find keys");
```

Comment:
nit: Where does this find keys? I'm guessing it's from the current user. Since this is library code, maybe we should rename this function from `new()` to e.g. `for_current_user()`? The idea is to make it clearer for anyone writing a server using `jujutsu_lib` that this is going to use a key from the user running the server (if that's what this does).